src/weights.jl

```diff
@@ -22,7 +22,18 @@ values(wv::AbstractWeights) = wv.values
 sum(wv::AbstractWeights) = wv.sum
 isempty(wv::AbstractWeights) = isempty(wv.values)
 
-Base.getindex(wv::AbstractWeights, i) = getindex(wv.values, i)
+function getindex(A::AbstractWeights{T}, i...) where {T}
+    # @boundscheck checkbounds(A, I...)
+    # Let Array indexing code handle everything
+    @inbounds r = A.values[i...]
+
+    if isa(r, Array)
+        return Weights(r)
```

Comment:
Should preserve the input weights vector type.

Reply:
What I mean is that e.g. a `FrequencyWeights` input should give a `FrequencyWeights` output.